packages/redux/src/subscriptions/reducer/subscriptionPackages.ts

```diff
@@ -1,98 +1,56 @@
 import * as actionTypes from '../actionTypes';
-import * as authenticationActionTypes from '../../users/authentication/actionTypes';
-import { AnyAction, combineReducers, Reducer } from 'redux';
+import type { AnyAction } from 'redux';
 import type { StoreState } from '../../types';
 import type { SubscriptionsState } from '../types';
 
-type SubscriptionPackagesState = SubscriptionsState['packages'];
+export const INITIAL_STATE: Record<string, SubscriptionsState['packages']> = {};
 
-export const INITIAL_STATE: SubscriptionPackagesState = {
-  error: null,
-  isLoading: false,
-  result: null,
-};
-
-const error = (state = INITIAL_STATE.error, action: AnyAction) => {
-  switch (action.type) {
-    case actionTypes.FETCH_SUBSCRIPTION_PACKAGES_FAILURE:
-      return action.payload.error;
-    case actionTypes.FETCH_SUBSCRIPTION_PACKAGES_REQUEST:
-      return INITIAL_STATE.error;
-    default:
-      return state;
-  }
-};
-
-const isLoading = (state = INITIAL_STATE.isLoading, action: AnyAction) => {
+const subscriptionPackagesReducer = (
+  state = INITIAL_STATE,
+  action: AnyAction,
+) => {
   switch (action.type) {
     case actionTypes.FETCH_SUBSCRIPTION_PACKAGES_REQUEST:
-      return true;
+      return {
+        ...state,
+        [action.meta.hash]: {
+          result: null,
+          isLoading: true,
+          error: null,
+        },
+      };
     case actionTypes.FETCH_SUBSCRIPTION_PACKAGES_SUCCESS:
+      return {
+        ...state,
+        [action.meta.hash]: {
+          result: action.payload.result,
+          isLoading: false,
+          error: null,
+        },
+      };
     case actionTypes.FETCH_SUBSCRIPTION_PACKAGES_FAILURE:
-      return false;
-    default:
-      return state;
-  }
-};
-
-const result = (state = INITIAL_STATE.result, action: AnyAction) => {
-  switch (action.type) {
-    case actionTypes.FETCH_SUBSCRIPTION_PACKAGES_SUCCESS:
-      return action.payload.result;
+      return {
+        ...state,
+        [action.meta.hash]: {
+          result: action.payload.result,
```

Comment:
```suggestion
          result: null,
```
Payload does not contain a result property when `FETCH_SUBSCRIPTION_PACKAGES_FAILURE` is dispatched